notebooks/aws-emr/init-configurations.json

```diff
@@ -0,0 +1,87 @@
+[
+{
+	"Classification":"spark",
+		"Properties":{
+			"enableSparkRapids":"true"
+		}
+},
+{
+	"Classification":"yarn-site",
+	"Properties":{
+		"yarn.nodemanager.resource-plugins":"yarn.io/gpu",
+		"yarn.resource-types":"yarn.io/gpu",
+		"yarn.nodemanager.resource-plugins.gpu.allowed-gpu-devices":"auto",
+		"yarn.nodemanager.resource-plugins.gpu.path-to-discovery-executables":"/usr/bin",
+		"yarn.nodemanager.linux-container-executor.cgroups.mount":"true",
+		"yarn.nodemanager.linux-container-executor.cgroups.mount-path":"/sys/fs/cgroup",
+		"yarn.nodemanager.linux-container-executor.cgroups.hierarchy":"yarn",
+		"yarn.nodemanager.container-executor.class":"org.apache.hadoop.yarn.server.nodemanager.LinuxContainerExecutor"
+	}
+},
+{
+	"Classification":"container-executor",
+	"Properties":{
+
+	},
+	"Configurations":[
+	{
+		"Classification":"gpu",
+		"Properties":{
+			"module.enabled":"true"
+		}
+	},
+	{
+		"Classification":"cgroups",
+		"Properties":{
+			"root":"/sys/fs/cgroup",
+			"yarn-hierarchy":"yarn"
+		}
+	}
+	]
+},
+{
+	"Classification":"spark-defaults",
+	"Properties":{
+		"spark.plugins":"com.nvidia.spark.SQLPlugin",
+		"spark.sql.sources.useV1SourceList":"",
+		"spark.executor.resource.gpu.discoveryScript":"/usr/lib/spark/scripts/gpu/getGpusResources.sh",
+		"spark.executor.extraLibraryPath":"/usr/local/cuda/targets/x86_64-linux/lib:/usr/local/cuda/extras/CUPTI/lib64:/usr/local/cuda/compat/lib:/usr/local/cuda/lib:/usr/local/cuda/lib64:/usr/lib/hadoop/lib/native:/usr/lib/hadoop-lzo/lib/native:/docker/usr/lib/hadoop/lib/native:/docker/usr/lib/hadoop-lzo/lib/native",
+		"spark.rapids.sql.concurrentGpuTasks":"2",
+		"spark.executor.resource.gpu.amount":"1",
+		"spark.executor.cores":"8",
+		"spark.task.cpus ":"1",
+		"spark.task.resource.gpu.amount":"1",
+		"spark.rapids.memory.pinnedPool.size":"2G",
+		"spark.executor.memoryOverhead":"2G",
+		"spark.sql.files.maxPartitionBytes":"256m",
+		"spark.sql.adaptive.enabled":"false",
+		"spark.python.worker.reuse":"true",
+		"spark.rapids.memory.gpu.minAllocFraction":"0.0001",
+		"spark.rapids.memory.gpu.pooling.enabled":"false",
+		"spark.rapids.sql.explain":"ALL",
+		"spark.rapids.memory.gpu.reserve":"20",
+		"spark.rapids.sql.python.gpu.enabled":"true",
+		"spark.rapids.memory.pinnedPool.size":"2G",
+		"spark.rapids.sql.batchSizeBytes":"512m",
+		"spark.locality.wait":"0",
+		"spark.sql.execution.sortBeforeRepartition":"false",
+		"spark.sql.execution.arrow.pyspark.enabled":"true",
+		"spark.sql.execution.arrow.maxRecordsPerBatch":"100000",
+		"spark.sql.cache.serializer com.nvidia.spark":"ParquetCachedBatchSerializer",
+		"spark.pyspark.python":"python3.9",
+		"spark.pyspark.driver.python":"python3.9",
+		"spark.dynamicAllocation.enabled":"false",
+		"spark.driver.memory":"20g",
+		"spark.rpc.message.maxSize":"512",
+		"spark.executorEnv.CUPY_CACHE_DIR":"/tmp/.cupy",
```

Comment:
Interesting.  Do we have this elsewhere?   What happens if it is not set?

Reply:
Yeah, this also appears in Dataproce. 

It not set, there will be a permission error writing to /home/.cupy.

Reply:
We'll have to look into whether isolation is needed for this directory in the case of concurrent spark jobs in the same cluster.

Reply:
Good question. The CUPY_CACHE_DIR comes from https://docs.cupy.dev/en/stable/reference/environment.html. 
Not sure if CUPY official handles concurrent cupy tasks. Need to try and find out. Created a ticket.